e2e/dashboards-suite/dashboard-restore.spec.ts

```diff
@@ -0,0 +1,55 @@
+import testDashboard from '../dashboards/TestRestoreDashboard.json';
+import { e2e } from '../utils';
+
+describe('Dashboard restore', () => {
+  beforeEach(() => {
+    e2e.flows.login(Cypress.env('USERNAME'), Cypress.env('PASSWORD'));
+  });
+
+  it('Deleted dashboards appear in Recently Deleted', () => {
+    e2e.flows.importDashboard(testDashboard, 1000, true);
+
+    e2e.pages.Dashboards.visit();
+
+    // Delete dashboard
+    deleteDashboard();
+
+    // Dashboard should appear in Recently Deleted
+    e2e.pages.RecentlyDeleted.visit();
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').should('exist');
+
+    // Restore dashboard
+    e2e.pages.Search.table.row('E2E Test - Restore Dashboard').find('[type="checkbox"]').click({ force: true });
+    cy.contains('button', 'Restore').click();
+    cy.contains('p', 'This action will restore 1 dashboard.').should('be.visible');
+    e2e.pages.ConfirmModal.delete().click();
+
+    // Dashboard should appear in Browse
+    e2e.pages.Dashboards.visit();
+    e2e.pages.BrowseDashboards.table.row('E2E Test - Restore Dashboard').should('exist');
+
```

Comment:
It's possible to delete dashboards in a different way. Can you please add a step here where you test the flow when deleting the dashboard from its settings page?